docker-app/qfieldcloud/subscription/tests/test_storage_quota.py

```diff
@@ -0,0 +1,169 @@
+import io
+import logging
+import os
+from datetime import date, timedelta
+
+from qfieldcloud.authentication.models import AuthToken
+from qfieldcloud.core import utils
+from qfieldcloud.core.models import Project, User
+from qfieldcloud.core.utils import list_versions
+from qfieldcloud.core.utils2.storage import delete_file_version
+from rest_framework import status
+from rest_framework.test import APITransactionTestCase
+
+from ..models import AccountType, ExtraPackage, ExtraPackageTypeStorage
+
+logging.disable(logging.CRITICAL)
+
+
+class QfcTestCase(APITransactionTestCase):
+    def _login(self, user):
+        token = AuthToken.objects.get_or_create(user=user)[0]
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {token}")
+
+    def _make_file(self, mb):
+        """Helper that returns a file of given size in megabytes"""
+        return io.BytesIO(os.urandom(1024 * int(mb * 1024)))
+
+    def test_storage_quota_calculation(self):
+        """This tests quotas"""
+
+        u1 = User.objects.create(username="u1")
+        p1 = Project.objects.create(name="p1", owner=u1)
+
+        default = AccountType.get_or_create_default()
+
+        # Initially, we have the default account type storage
+        self.assertEqual(u1.useraccount.storage_quota_left, default.storage_mb)
+
+        # Changing account type changes the quota
+        micro = AccountType.objects.create(code="micro", storage_mb=1)
+        u1.useraccount.account_type = micro
+        u1.useraccount.save()
+        self.assertEqual(u1.useraccount.storage_quota_left, 1)
+
+        # Adding an extra package increses the quota
+        ep_med = ExtraPackageTypeStorage.objects.create(code="extra_med", megabytes=2)
```

Comment:
```suggestion
        extra_2mb = ExtraPackageTypeStorage.objects.create(code="extra_2mb", megabytes=2)
```
Sorry, these cryptic names are already confusing for me.